apps/docs-app/docs/experiments/sfc/index.md

```diff
@@ -0,0 +1,344 @@
+
+# Inputs and Outputs
+
+To add inputs and outputs to an Analog component, you use the new Angular signals API.
+
+Let's explore what that looks like in practical terms.
+
+## Inputs
+
+Inputs can be added to a component or directive in the Analog format using [the new `input` signal API](https://angular.io/guide/signal-inputs):
+
+```typescript
+const namedInput = input();
+```
+
+This will add an input with the name of `namedInput` that can be used in the template like so:
+
+```html
+<template>
+  <SomeComponent [namedInput]="someValue" />
+</template>
+```
+
+## Outputs
+
+Outputs are added in the Analog format like so:
+
+```html
+<script lang="ts">
+  const selectItem = new EventEmitter();
+</script>
+```
+
+The above will be transformed to:
+
+```typescript
+class Component {
+  @Output() selectItem = new EventEmitter();
+}
+```
+
+And can be used in the template like so:
+
+```html
+<template>
+  <SomeComponent (selectItem)="doSomething($event)" />
+</template>
+```
+
+> In the future, this will be replaced with [the `output` signals API](https://blog.angular.io/meet-angulars-new-output-api-253a41ffa13c).
+
+## Models
+
+The new [`model` signal API](https://angular.io/api/core/model) is not yet supported.
+
+# Dedicated Template and Style Files
+
+If you like the developer experience of Analog's `<script>` to build your logic, but don't want your template and styling in the same file, you can break those out to their own files using:
+
+- `templateUrl`
+- `styleUrl`
+- `styleUrls`
+
+In `defineMetadata`, like so:
+
+```html
+<script lang="ts">
+  defineMetadata({
+    selector: 'app-root',
+    templateUrl: './test.html',
+    styleUrl: './test.css',
+  });
+
+  onInit(() => {
+    alert('Hello World');
+  });
+</script>
+```
+
+# Directives
+
+Any `.analog` file without a `<template>` tag or usage of `templateUrl` in the `defineMetadata` function are treated as Angular Directives.
+
+Here's an example of a directive that focuses an input and has two lifecycle methods:
+
+```html
+<script lang="ts">
+  import { inject, ElementRef, afterNextRender } from '@angular/core';
+
+  defineMetadata({
+    selector: 'input[directive]',
+  });
+
+  const elRef = inject(ElementRef);
+
+  afterNextRender(() => {
+    elRef.nativeElement.focus();
+  });
+
+  onInit(() => {
+    console.log('init code');
+  });
+
+  effect(() => {
+    console.log('just some effect');
+  });
+</script>
+```
+
+# Markdown
+
+If you'd like to write Markdown as your template rather than Angular-enhanced HTML, you can add `lang="md"` to your `<template>` tag in an `.analog` file:
+
+```html
+<template lang="md"> # Hello World </template>
+```
+
+This can be used in combination with the other SFC tags: `<script>` and `<style>`.
+
+# Limitations
+
+There are a few limitations to the Analog format:
+
+- You cannot use decorator APIs (`@Input`, `@Component`, `@ViewChild`)
+- You must have `lang="ts"` present in the `<script>`
```

Comment:
Question: Why do we consider this to be a limitation?

iirc, `lang="ts"` is the default anyway when not provided. It is needed as a workaround for the lack of editor/IDE support. Are you suggesting that being forced to only use TypeScript is the limitation?

Reply:
I could be wrong, but according to the source code, it looks like `lang="ts"` is a requirement and is not optional:

https://github.com/analogjs/analog/blob/beta/packages/vite-plugin-angular/src/lib/authoring/constants.ts#L10

I'm marking it as a limitation here for two reasons:

1) It's a bit of a "gotcha" for those unfamiliar with Analog and only familiar with HTML
2) Vue authors looking to experiment with Angular may not be familiar with this "restriction" to only use TypeScript and, given the authoring similarities, I think is a good think to highlight for those looking to explore the Angular space